src/python/WMCore/WMRuntime/Watchdog.py

```diff
@@ -78,6 +81,34 @@ def setupMonitors(self, task, wmbsJob):
                 #This should be a config section
                 monitorArgs = getattr(task.data.watchdog, monitor)
                 args = monitorArgs.dictionary_()
+                # Scale resources according to the HTCondor runtime environment.
+                origCores = 1
+                for stepName in task.listAllStepNames():
+                    sh = task.getStepHelper(stepName)
+                    origCores = max(origCores, sh.getNumberOfCores())
+                resources = {'cores': origCores}
+                origMaxRSS = args.get('maxRSS')
+                if origMaxRSS:
+                    origMaxRSS = int(origMaxRSS / 1024.)  # HTCondor expects MB; we get KB.
+                    resources['memory'] = origMaxRSS
+                # Actually parses the HTCondor runtime
+                resizeResources(resources)
+                # We decided to only touch Watchdog settings if the number of cores changed.
+                # (even if this means the watchdog memory is wrong for a slot this size).
+                changedCores = origCores != resources['cores']
+                # HTCondor doesn't explicitly scale VSize; it's also not clear what
+                # resources this manages (as we already watch the memory use) or how
+                # it should relate to other resources (such as memory or cores used).
+                # Hence, we simply remove it if we change anything about the memory.
+                # If we did base maxRSS off the memory in the HTCondor slot, subtract a bit
+                # off the top so watchdog triggers before HTCondor does.
+                if changedCores and origMaxRSS and (resources['memory'] != origMaxRSS):
+                    args.pop('maxVSize', None)
+                    args['maxRSS'] = 1024 * (resources['memory'] - 50)  # Convert back to KB
+
+                logging.info("Watchdog modified: %s. Final settings:", changedCores)
+                for k, v in args.iteritems():
+                    logging.info("  %s: %r", k, v)
             # Actually initialize the monitor variables
             mon.initMonitor(task = task, job = wmbsJob,
```

Comment:
It seems that Dashboard gets its cores count from `src/python/WMCore/WMRuntime/DashboardInterface.py` which, in turn, pull the information from the task description (the `task` parameter here).

Unfortunately, the task description doesn't get its core count updated until later (`SetupCMSSWPset.py`).

I think we could set, after L107, an override for the number of cores.  This would be passed through `DashboardMonitor.py` to the class in `DashboardInterface.py` and override the `maxCores` member.